### PR TITLE
Feature/django v4

### PR DESCRIPTION
**Pull request by martin-marty**
As per the promise, github workflow to run tests

### CODE REVIEW

**Comment by chris-umed** [comment]
@sjamaan - this PR adds automated test execution using Github actions and drops support for unsupported versions of Django allowing a simple upgrade path to Django 4. 
Would be great if you can either provide us the ability to review, merge and release a new version or do it on this PR?

**Comment by uhurusurfa** [comment]
@sjamaan - this PR adds automated test execution using Github actions and drops support for unsupported versions of Django allowing a simple upgrade path to Django 4.
Would be great if you can either provide us the ability to review, merge and release a new version or do it on this PR?

**Comment by sjamaan** [comment]
As I mentioned in the related issue, I defer to @abzainuddin or someone else from Code Yellow, as they use this for production code. At a quick glance, the changes look fine, except that they completely seem to drop entire ranges of Django and Python versions.

I don't know if Code Yellow is running any code that hasn't yet upgraded to the supported Django version range, so I really don't want to "blindly" accept the PR even though the code is technically fine. Is there a reason the older Django versions are dropped? I don't see any code changes that would break it. Just because the Django project doesn't officially support these versions anymore doesn't mean compatibility should be artificially slashed, IMO.

If there's a technical reason to make supporting those older versions unnecessarily hard, that's a different story of course.

File: .github/workflows/develop.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+name: Relativedleta tests
```
Comment (sjamaan) [inline .github/workflows/develop.yml:1]: Typo :)

**Comment by abzainuddin** [comment]
I'll ask if someone can take a look next week (lots of holidays atm).

**Comment by uhurusurfa** [comment]
@sjamaan @abzainuddin  - to see the Github actions tests that are executed you will need to look in the upstream repo (https://github.com/martin-marty/django-relativedelta/actions/runs/2846106086) to see the results as the actions workflows will only run for PR's from external repos once merged into this repo:
https://github.com/orgs/community/discussions/25012

**Comment by uhurusurfa** [comment]
@abzainuddin - any chance this can be merged soon? We are keen to upgrade our platform to Django4 ASAP. As you can see, the unit tests now run as part of the CI/CD pipeline and provide a decent amount of testing for the code.

**Comment by chris-umed** [comment]
@abzainuddin  - sorry to be a pest but could you look into merging this please?
It would be very much appreciated.
The unit tests now run as part of the PR and provide a high level of certainty that the app works as designed given the relatively extensive unit tests so really does not need a significant amount of time to review, merge and release to PyPi

**Comment by robinzelders** [comment]
done

**Pull request merged**